projects/js-packages/charts/src/components/bar-chart/bar-chart.tsx

```diff
@@ -88,6 +106,37 @@ const BarChart: FC< BarChartProps > = ( {
 		[ chartOptions.tooltip ]
 	);
 
+	const renderPattern = useCallback(
+		( index: number, color: string ) => {
+			const patternType = index % 4;
+			const commonProps = {
+				id: getPatternId( chartId, index ),
+				stroke: color,
+				strokeWidth: 1,
+			};
+
+			switch ( patternType ) {
+				case 0:
+				default:
+					return (
+						<PatternLines
+							{ ...commonProps }
+							width={ 5 }
+							height={ 5 }
+							orientation={ [ 'diagonal' ] }
+						/>
+					);
+				case 1:
+					return <PatternCircles { ...commonProps } width={ 6 } height={ 6 } fill={ color } />;
+				case 2:
+					return <PatternWaves { ...commonProps } width={ 4 } height={ 4 } />;
+				case 3:
+					return <PatternHexagons { ...commonProps } size={ 8 } height={ 3 } />;
```

Comment:
These are the values from the examples: https://airbnb.io/visx/docs/pattern

On very skinny bars it's tough to make the patterns display well. We could consider allowing rendering a custom pattern or adjusting the size of these via props.

Reply:
The https://airbnb.io/visx/docs/pattern#Path component looks like a way to render a custom SVG shape using the `path` attribute. We could allow this to be configured via the theme maybe.